src/modules/logicModule/LM_PC_FundingPot_v1.sol

```diff
@@ -0,0 +1,1438 @@
+
+    /// @inheritdoc ILM_PC_FundingPot_v1
+    function createRound(
+        uint roundStart_,
+        uint roundEnd_,
+        uint roundCap_,
+        address hookContract_,
+        bytes memory hookFunction_,
+        bool autoClosure_,
+        AccumulationMode accumulationMode_
+    ) external onlyModuleRole(FUNDING_POT_ADMIN_ROLE) returns (uint32) {
+        unchecked {
+            roundCount++;
+        }
+
+        uint32 roundId = roundCount;
+
+        Round storage round = rounds[roundId];
+        _setAndValidateRoundParameters(
+            round,
+            roundStart_,
+            roundEnd_,
+            roundCap_,
+            hookContract_,
+            hookFunction_,
+            autoClosure_,
+            accumulationMode_
+        );
+
+        emit RoundCreated(
+            roundId,
+            roundStart_,
+            roundEnd_,
+            roundCap_,
+            hookContract_,
+            hookFunction_,
+            autoClosure_,
+            accumulationMode_
+        );
+
+        return uint32(roundId);
+    }
+
+    /// @inheritdoc ILM_PC_FundingPot_v1
+    function editRound(
+        uint32 roundId_,
+        uint roundStart_,
+        uint roundEnd_,
+        uint roundCap_,
+        address hookContract_,
+        bytes memory hookFunction_,
+        bool autoClosure_,
+        AccumulationMode accumulationMode_
+    ) external onlyModuleRole(FUNDING_POT_ADMIN_ROLE) {
+        Round storage round = rounds[roundId_];
+
+        _validateEditRoundParameters(round);
+
+        _setAndValidateRoundParameters(
+            round,
+            roundStart_,
+            roundEnd_,
+            roundCap_,
+            hookContract_,
+            hookFunction_,
+            autoClosure_,
+            accumulationMode_
+        );
+
+        emit RoundEdited(
+            roundId_,
+            roundStart_,
+            roundEnd_,
+            roundCap_,
+            hookContract_,
+            hookFunction_,
+            autoClosure_,
+            accumulationMode_
+        );
+    }
+
+    /// @inheritdoc ILM_PC_FundingPot_v1
+    function setAccessCriteria(
+        uint32 roundId_,
+        uint8 accessCriteriaType_,
+        uint8 accessCriteriaId_, // Optional: 0 for new, non-zero for edit
+        address nftContract_,
+        bytes32 merkleRoot_,
+        address[] calldata allowedAddresses_
+    ) external onlyModuleRole(FUNDING_POT_ADMIN_ROLE) {
+        Round storage round = rounds[roundId_];
+
+        if (accessCriteriaType_ > MAX_ACCESS_CRITERIA_TYPE) {
+            revert Module__LM_PC_FundingPot__InvalidAccessCriteriaId();
+        }
+
+        _validateEditRoundParameters(round);
+
+        uint8 criteriaId;
+        bool isEdit = false;
+
+        // If accessCriteriaId_ is 0, create a new access criteria
+        // Otherwise, edit the existing one
+        if (accessCriteriaId_ == 0) {
+            unchecked {
+                criteriaId = ++roundIdToNextAccessCriteriaId[roundId_];
```

Comment:
if you have prblems witht he contract size, a small gain is to have a global counter for teh criteriaId instead of a spearate one for each round.